zitadel/v2/helper/form.go

```diff
@@ -76,9 +77,18 @@ func formFilePost(clientInfo *ClientInfo, endpoint, path string, additionalHeade
 		r.Header.Add(k, v)
 	}
 
-	client, err := NewClientWithInterceptor(clientInfo.Issuer, clientInfo.KeyPath, []string{oidc.ScopeOpenID, zitadel.ScopeZitadelAPI()})
-	if err != nil {
-		return diag.Errorf("failed to create client: %v", err)
+	if clientInfo.KeyPath != "" {
+		client, err = NewClientWithInterceptorFromKeyFile(clientInfo.Issuer, clientInfo.KeyPath, []string{oidc.ScopeOpenID, zitadel.ScopeZitadelAPI()})
+		if err != nil {
+			return diag.Errorf("failed to create client: %v", err)
+		}
+	} else if len(clientInfo.Data) > 0 {
+		client, err = NewClientWithInterceptorFromKeyFile(clientInfo.Issuer, clientInfo.KeyPath, []string{oidc.ScopeOpenID, zitadel.ScopeZitadelAPI()})
```

Comment:
Shouldn't that be NewClientWithInterceptorFromKeyFileData ?

Reply:
Ops sorry. I'll fix that